ledger-service/http-json/src/main/scala/com/digitalasset/http/util/Concurrent.scala

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http.util
+
+object Concurrent {
+
+  /** Equal to a Semaphore with max count equal to one. */
+  final case class Mutex() extends java.util.concurrent.Semaphore(1, true)
+}
```

Comment:
I don't think we need this, especially as a publicly available class.

A simple
```
private[this] def newMutex() = new java.util.concurrent.Semaphore(1, true)
```
in `PackageService` will do.

I'm also trying to have a better look at the code that uses this, as I'd like to understand why `synchronized` won't do.

Reply:
I agree, in the past there was a bit more stuff in here but the code changed again. Thanks for the suggestion :)